hibernate-core/src/main/java/org/hibernate/tuple/entity/CompositeGeneratorBuilder.java

```diff
@@ -193,7 +193,7 @@ public Object generate(SharedSessionContractImplementor session, Object owner, O
 			else {
 				for ( int i = 0; i < size; i++ ) {
 					final Generator generator = generators.get( i );
-					if ( generator != null ) {
+					if ( generator != null && generator.getEventTypes().contains( eventType ) ) {
```

Comment:
When we have a composite generator that consists of a mix of generators with different event types, we are triggering the generator for all the events, but the generators might be inapplicable for some of them ... 

@gavinking does that fix make sense to you? Or should we approach it differently?

Reply:
@marko-bekhta LGTM, but this is really more of a question for @mbladel I guess.

Reply:
oh ok, thanks 😃 👍🏻 I saw you recently cleaned up that file, so thought to ping you 🙈